src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -1,14 +1,127 @@
 package core.basesyntax.service;
 
+import core.basesyntax.NoValidDataException;
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int VALID_LOGIN_LENGTH = 6;
+    private static final int VALID_PASSWORD_LENGTH = 6;
+    private static final int VALID_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        validator(user);
+        return storageDao.add(user);
+    }
+
+    private void validator(User user) {
+        checkIfUserHasNoData(user);
+        checkIfUserAlreadyExist(user);
+        validateLogin(user);
+        validatePassword(user);
+        validateAge(user);
+    }
+
+    private void checkIfUserHasNoData(User user) {
+        if (user == null) {
+            throw new NoValidDataException("User can't be added");
```

Comment:
Perhaps, It will be better to clarify, what's the reason of the exception, not just "User can't be added"